nixos/modules/i18n/input-method/default.nix

```
@@ -28,8 +30,10 @@ in
 {
   options.i18n = {
     inputMethod = {
-      enabled = mkOption {
-        type    = types.nullOr (types.enum [ "ibus" "fcitx5" "nabi" "uim" "hime" "kime" ]);
+      enabled = mkEnableOption "an additional input method type";
```

```suggestion
      enable = mkEnableOption "an additional input method type";
```
for consistency with any other module.
Then we also add a "removed option" assertion

---

I've made both of these changes, but they seem to be mutually exclusive, and the two tests are now failing.

Am I just doing it wrong? I'd appreciate your guidance.

Test with new style config (`nix-build -A nixosTests.installed-tests.ibus`), where the error is in the module definition itself:
```
error:
       Failed assertions:
       - The option definition `i18n.inputMethod.enabled' in `/home/gy/src/nixpkgs/nixos/modules/i18n/input-method/default.nix' no longer has any effect; please remove it.
       Use <literal>i18n.inputMethod.type and .enable = true; instead<literal>

```

Test with old style config (`nix-build -A nixosTests.fcitx5`):
```
error: The option `i18n.inputMethod.enabled' can no longer be used since it's been removed. Use <literal>i18n.inputMethod.type and .enable = true; instead<literal>
```

---

You're right. We should not use mk(Removed|Renamed)OptionModule here.
Removing that fixes the tests.

What we could do (to ensure that users get a warning with `.enabled`) is to set `config.warnings` if `.enabled != null`.